File: app/src/main/java/com/github/goldy1992/mp3player/service/library/content/filter/SongsFromFolderResultsFilter.java
package com.github.goldy1992.mp3player.service.library.content.filter;

import android.support.v4.media.MediaBrowserCompat.MediaItem;

import androidx.annotation.NonNull;

import com.github.goldy1992.mp3player.commons.MediaItemUtils;

import java.io.File;
import java.util.Iterator;
import java.util.List;

import javax.inject.Inject;

public class SongsFromFolderResultsFilter implements ResultsFilter {


    @Inject
    public SongsFromFolderResultsFilter() {}


    @Override
    public List<MediaItem> filter(@NonNull String query,
                                  @NonNull List<MediaItem> results) {
        File queryPath = new File(query);
        Iterator<MediaItem> iterator = results.listIterator();

        while(iterator.hasNext()) {
            MediaItem currentItem = iterator.next();
            String directoryPath = MediaItemUtils.getDirectoryPath(currentItem);
            if (directoryPath == null || !directoryPath.equalsIgnoreCase(queryPath.getAbsolutePath().toUpperCase())) {
                iterator.remove();
            }
        }
        return results;
    }
}

File: service/src/main/java/com/github/goldy1992/mp3player/service/library/content/filter/SongsFromFolderResultsFilter.kt
package com.github.goldy1992.mp3player.service.library.content.filter

import androidx.media3.common.MediaItem
import com.github.goldy1992.mp3player.commons.MediaItemUtils.getDirectoryPath
import java.io.File
import java.util.Locale
import javax.inject.Inject

class SongsFromFolderResultsFilter @Inject constructor() : ResultsFilter {
    override fun filter(query: String,
                        results: MutableList<MediaItem>?): List<MediaItem> {
        val queryPath = File(query)

        return results!!.filter {
            val directoryPath = getDirectoryPath(it)
            directoryPath != null && directoryPath.equals(queryPath.absolutePath.uppercase(Locale.ROOT), ignoreCase = true)
        }
    }
}